src/app/components/rejestracja/rejestracja.component.html

```diff
@@ -6,32 +6,70 @@
+          <ng-template #elseBlock1>
+            <span *ngIf="this.rejestracjaForm.controls['haslo'].errors.minlength">Za krótkie hasło.</span>
+          </ng-template>
+        </mat-error>
       </mat-form-field>
       <mat-form-field class="example-full-width">
-        <input matInput type="password" placeholder="Powtórz Hasło">
+        <input matInput type="password" placeholder="Powtórz Hasło" required formControlName="haslo2" (change)="weryfikujHaslo()">
         <mat-hint>Potwierdź hasło!</mat-hint>
+        <mat-error *ngIf="this.rejestracjaForm.controls['haslo2'].invalid">
+          <span *ngIf="this.rejestracjaForm.controls['haslo2'].errors.czyHasloJestTakieSamo; else elseBlock2">Podane hasło różni się od podanego wcześniej.</span>
+          <ng-template #elseBlock2>
+            <span *ngIf="this.rejestracjaForm.controls['haslo2'].errors.required">Musisz potwierdzić hasło.</span>
+          </ng-template>
+        </mat-error>
       </mat-form-field>
       <mat-form-field class="example-full-width">
         <input matInput placeholder="Adres e-mail" formControlName="email">
         <mat-hint>Wpisz e-mail!</mat-hint>
+        <mat-error *ngIf="this.rejestracjaForm.controls['email'].invalid">
+          <span *ngIf="this.rejestracjaForm.controls['email'].errors.required; else elseBlock3">Musisz podać maila.</span>
+          <ng-template #elseBlock3>
+            <span *ngIf="this.rejestracjaForm.controls['email'].errors.pattern">Zły format maila.</span>
+          </ng-template>
+        </mat-error>
       </mat-form-field>
       <mat-form-field class="example-full-width">
         <input matInput placeholder="Imię" formControlName="imie">
         <mat-hint>Wpisz imię!</mat-hint>
+        <mat-error *ngIf="this.rejestracjaForm.controls['imie'].invalid">
+          <span *ngIf="this.rejestracjaForm.controls['imie'].errors.required; else elseBlock4">Musisz podać imię.</span>
+          <ng-template #elseBlock4>
+            <span *ngIf="this.rejestracjaForm.controls['imie'].errors.pattern">Niepoprawne imię.</span>
+          </ng-template>
+        </mat-error>
       </mat-form-field>
       <mat-form-field class="example-full-width">
         <input matInput placeholder="Nazwisko" formControlName="nazwisko">
         <mat-hint>Wpisz nazwisko!</mat-hint>
+        <mat-error *ngIf="this.rejestracjaForm.controls['nazwisko'].invalid">
+          <span *ngIf="this.rejestracjaForm.controls['nazwisko'].errors.required; else elseBlock5">Musisz podać nazwisko.</span>
+          <ng-template #elseBlock5>
+            <span *ngIf="this.rejestracjaForm.controls['nazwisko'].errors.pattern">Niepoprawne nazwisko.</span>
+          </ng-template>
+        </mat-error>
       </mat-form-field>
       <mat-form-field class="example-full-width">
         <input matInput placeholder="Data urodzenia" formControlName="dataUrodzenia">
```

Comment:
Proponuję dodać wybór daty z kalendarza tak zrobiono to w formularzu dodawania oferty. Format daty powinien być dd.mm.rrrr.